os/fs/driver/mtd/smart.c

```diff
@@ -1862,22 +1870,24 @@ static int smart_scan(FAR struct smart_struct_s *dev)
 		if (logicalsector > 0 && header.status != CONFIG_SMARTFS_ERASEDSTATE && !status_released && status_committed) {
 
 			/* Map the sector and update the free sector counts. */
-			if (header.seq >= sector_seq_log[logicalsector]) {
+			if (logicalsector < dev->totalsectors) {
```

Comment:
you can add this condition beside of above state (logicalsecotr > 0 && logicalsector < dev->totalsectors)
But, most of all these code will be removed soon, you raised this PR to um..just clean current version?

Reply:
I'll make the change as you suggested.
And yes, these changes only clean the current code a little bit, we have discussed the rest and I won't try to modify that as of now.